core/src/main/java/org/mule/processor/strategy/AbstractThreadingProfileProcessingStrategy.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.processor.strategy;
+
+import org.mule.api.MuleContext;
+import org.mule.api.config.ThreadingProfile;
+import org.mule.api.construct.FlowConstruct;
+import org.mule.api.context.WorkManager;
+import org.mule.api.lifecycle.Startable;
+import org.mule.api.lifecycle.Stoppable;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.api.processor.MessageProcessorChainBuilder;
+import org.mule.api.processor.ProcessingStrategy;
+import org.mule.api.processor.StageNameSource;
+import org.mule.config.ChainedThreadingProfile;
+import org.mule.processor.AsyncInterceptingMessageProcessor;
+import org.mule.util.concurrent.ThreadNameHelper;
+import org.mule.work.MuleWorkManager;
+
+import java.util.List;
+
+public abstract class AbstractThreadingProfileProcessingStrategy implements ProcessingStrategy
+{
+
+    protected Integer maxThreads;
+    protected Integer minThreads;
+    protected Integer maxBufferSize;
+    protected Long threadTTL;
+    protected Long threadWaitTimeout;
+    protected Integer poolExhaustedAction;
+    protected ProcessingStrategy synchronousProcessingStrategy = new SynchronousProcessingStrategy();
+
+
+    protected ThreadingProfile createThreadingProfile(MuleContext muleContext)
```

Comment:
I didn't change format to not add noise to PR as this was just moved.  Can update if required.  You are referring to lack of braces, right?